### PR TITLE
DEVPROD-69 PutMany for minutely jobs

### PR DESCRIPTION
**Pull request by ybrill**
[DEVPROD-69](https://jira.mongodb.org/browse/DEVPROD-69)

### Description
This is the same as what we did with the 15 seconds jobs in https://github.com/evergreen-ci/evergreen/pull/6917. I'm particularly excited to see if there's an effect on the amboy db's performance as a result of batching the insertion of jobs to process events.

### Testing
In staging a task sent a notification when it finished.

### CODE REVIEW

File: units/crons_remote_minute.go [hunk 1]
```diff
@@ -45,55 +45,47 @@ func (j *cronsRemoteMinuteJob) Run(ctx context.Context) {
...
-	}
-
-	eventNotifierQueue, err := j.env.RemoteQueueGroup().Get(appCtx, eventNotifierQueueGroup)
-	if err != nil {
-		catcher.Wrap(err, "getting event notifier queue")
-	} else {
-		catcher.Add(PopulateEventNotifierJobs(j.env)(ctx, eventNotifierQueue))
-	}
+	// Add generate tasks fallbacks to their versions' queues.
+	catcher.Add(enqueueFallbackGenerateTasksJobs(appCtx, ctx, j.env, ts))
```
Comment (julianedwards) [inline units/crons_remote_minute.go:88]: Can this logic not be separated out similar to the pattern above where the jobs are returned and L78 takes care of doing the bulk enqueue?

**Review by julianedwards — CHANGES_REQUESTED** [review]

File: units/event_notifier.go [hunk 2]
```diff
@@ -120,7 +119,9 @@ func (j *eventNotifierJob) processEvent(ctx context.Context, e *event.EventLogEn
 		catcher.AddWhen(shouldLogError, errors.Wrap(err, "bulk inserting notifications"))
 	}
 
-	catcher.Add(dispatchNotifications(ctx, n, j.q, j.flags))
+	jobs, err := notificationJobs(ctx, n, j.flags, utility.RoundPartOfMinute(0))
+	catcher.Add(errors.Wrap(err, "getting notification jobs"))
```
[the remark this answers is not in the document]
Comment (ybrill) [reply units/event_notifier.go:124]: done

File: units/crons.go [hunk 3]
```diff
@@ -1282,13 +1227,15 @@ func PopulatePodResourceCleanupJobs() amboy.QueueOperation {
 	}
 }
 
-// dispatchUnprocessedNotifications gets unprocessed notifications
-// leftover by previous runs and dispatches them
-func dispatchUnprocessedNotifications(ctx context.Context, q amboy.Queue, flags *evergreen.ServiceFlags) error {
-	unprocessedNotifications, err := notification.FindUnprocessed()
+func populateQueueGroup(appCtx, ctx context.Context, env evergreen.Environment, queueGroupName string, factory cronJobFactory, ts time.Time) error {
```
[the remark this answers is not in the document]
Comment (ybrill) [reply units/crons.go:1230]: done

File: units/crons.go [hunk 4]
```diff
@@ -221,156 +201,135 @@ func PopulateTaskMonitoring(mins int) amboy.QueueOperation {
...
+		return nil, errors.Wrap(err, "getting admin settings")
 	}
-}
+	containerPools := settings.ContainerPools.Pools
 
-func PopulateContainerStateJobs(env evergreen.Environment) amboy.QueueOperation {
-	return func(ctx context.Context, queue amboy.Queue) error {
-		catcher := grip.NewBasicCatcher()
-		ts := utility.RoundPartOfHour(1).Format(TSFormat)
+	// Create ParentDecommissionJob for each distro
```
[the remark this answers is not in the document]
Comment (ybrill) [reply units/crons.go]: added

File: units/crons.go [hunk 5]
```diff
@@ -221,156 +201,135 @@ func PopulateTaskMonitoring(mins int) amboy.QueueOperation {
...
+		return nil, errors.Wrap(err, "finding all parent hosts")
+	}
 
-		// Create oldestImageJob when images take up too much disk space
-		for _, p := range parents {
-			catcher.Wrapf(amboy.EnqueueUniqueJob(ctx, queue, NewOldestImageRemovalJob(&p, evergreen.ProviderNameDocker, ts)), "enqueueing oldest image removal job for parent host '%s'", p.Id)
-		}
-		return errors.Wrap(catcher.Resolve(), "populating oldest image removal jobs")
+	var jobs []amboy.Job
+	// Create oldestImageJob when images take up too much disk space
```
[the remark this answers is not in the document]
Comment (ybrill) [reply units/crons.go]: added

File: units/crons.go [hunk 6]
```diff
@@ -221,156 +201,135 @@ func PopulateTaskMonitoring(mins int) amboy.QueueOperation {
...
-		}
+func containerStateJobs(ctx context.Context, ts time.Time) ([]amboy.Job, error) {
+	parents, err := host.FindAllRunningParents(ctx)
+	if err != nil {
+		return nil, errors.Wrap(err, "finding parent hosts")
+	}
 
-		return catcher.Resolve()
+	var jobs []amboy.Job
+	// create job to check container state consistency for each parent
```
[the remark this answers is not in the document]
Comment (ybrill) [reply units/crons.go]: ✅

File: units/crons_remote_minute.go (hunk 1 above)
Comment (ybrill) [reply units/crons_remote_minute.go]: The reason this was left out was that every version's generate tasks insert in their own queue ([here](https://github.com/evergreen-ci/evergreen/blob/2d7b487e108f2d2ae793897cab22e46c3b10c9d5/units/generate_tasks.go#L292)).

Comment (julianedwards) [reply units/crons_remote_minute.go]: Ah missed that line, makes sense then!

**Review by julianedwards — APPROVED** [review]

**Pull request merged**